sorted_db/db.go

```diff
@@ -92,17 +101,18 @@ func (db *DB) close() {
 		db.f = nil
 	}
 	db.size = -1
+	return
 }
 
 // Remap DB to the same backing file
 func (db *DB) Remap() error {
-	db.RLock()
+	db.mutex.RLock()
```

Comment:
I was following that pattern when writing `RangeMatch` but I had wanted to circle back and ask about it because I get the sense that using `defer` in `Remap`, `Search`, `ForwardMatch`, and `RangeMatch` would help guarantee that locks don't ever leak.